src/components/modals/MigrateAccounts/steps/03-step-currency.js

```diff
@@ -0,0 +1,193 @@
+// @flow
+
+import invariant from 'invariant'
+import React, { PureComponent, Fragment } from 'react'
+import logger from 'logger'
+import { Trans } from 'react-i18next'
+
+import type { Account } from '@ledgerhq/live-common/lib/types/account'
+import { addAccounts } from '@ledgerhq/live-common/lib/account'
+import { DeviceShouldStayInApp } from '@ledgerhq/errors'
+import Text from 'components/base/Text'
+import TrackPage from 'analytics/TrackPage'
+
+import Button from 'components/base/Button'
+import Box from 'components/base/Box'
+import { CurrencyCircleIcon } from 'components/base/CurrencyBadge'
+import { getCurrencyBridge } from 'bridge'
+import type { StepProps } from '../index'
+
+type Props = StepProps & { replaceAccounts: (Account[]) => void }
+
+// $FlowFixMe
+const remapTransportError = (err: mixed, appName: string): Error => {
+  if (!err || typeof err !== 'object') return err
+
+  const { name, statusCode } = err
+
+  const errorToThrow =
+    name === 'BtcUnmatchedApp' || statusCode === 0x6982 || statusCode === 0x6700
+      ? new DeviceShouldStayInApp(null, { appName })
+      : err
+
+  return errorToThrow
+}
+
+class StepCurrency extends PureComponent<Props> {
+  componentDidMount() {
+    this.props.setScanStatus('scanning')
+    this.startScanAccountsDevice()
+  }
+
+  componentWillUnmount() {
+    this.unsub()
+  }
+
+  scanSubscription = null
+
+  unsub = () => {
+    if (this.scanSubscription) {
+      this.scanSubscription.unsubscribe()
+    }
+  }
+
+  startScanAccountsDevice() {
+    this.unsub()
+    const {
+      currency,
+      device,
+      setScanStatus,
+      addMigratedAccount,
+      replaceAccounts,
+      transitionTo,
+    } = this.props
+    try {
```

Comment:
I know this was taken from the scan accounts but it's getting a bit odd legacy stuff.
you can remove the try catch because unecessary and you can just `if(!currency||!device) return`